### DISCUSSION TITLE
advanced example

### DISCUSSION THREAD

**Discussion by Tomsaer** [RehanSaeed/Schema.NET#9] [Q&A]
Hi, can someone please write a example how to create this structure? Can't find any advanced examples.

Thank you

<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "SearchAction",
  "query": "phone",
  "result": {
    "@type": "ItemList",
    "name": "Search Results",
    "itemListElement": [
      {
        "@type": ["ListItem", "Product"],
        "position": "1",
        "name": "Android Phone PX100",
        "url": "https://myshop.com/products/236",
        "offers": {
          "@type": "Offer",
          "priceCurrency": "EUR",
          "price": "120"
        }
      },
      {
        "@type": ["ListItem", "Product"],
        "position": "2",
        "name": "iPhone X",
        "url": "https://myshop.com/products/293",
        "offers": {
          "@type": "Offer",
          "priceCurrency": "EUR",
          "price": "999"
        }
      }
    ]
  },
  "instrument": [{
    "@type": "ChooseAction",
    "name": "Sort by",
    "actionOption": "Relevance"
  },
  {
    "@type": "ChooseAction",
    "name": "Color",
    "actionOption": ["Black", "Silver"]
  }]
}
</script>

**Comment by RehanSaeed** [comment]
There are several examples in the unit tests [here](https://github.com/RehanSaeed/Schema.NET/tree/master/Tests/Schema.NET.Test) but not specifically for the `SearchAction` type. Please feel free to add one if you find the time.
